### ISSUE TITLE
UI is stuck on the "Loading..." screen

### ISSUE THREAD

**Issue by jpaxtonrs** [reportportal/reportportal#2061]
**Describe the bug**
When I open the browser and navigate to ReportPortal's UI, it is stuck in the `Loading...` screen.

**Expected behavior**
It should direct the user to the login screen prompting for user credentials.

**Additional context**
I am deploying ReportPortal into AWS `Fargate` using `CloudFormation`.
Here is a breakdown of how I am deploying the infrastructure:
- `namespace`
  - This CloudFormation Stack is responsible for deploying out Namespaces for Service Discovery so that the `Fargate` containers can communicate with each other and store their hostnames in the Container's Environment Variable prior to deployment. For example, the `api` Container needs to store `RP_AMQP_HOST` variable, which means it needs to know the `rabbitmq` host before deployment.
- `efs`
  - This CloudFormation Stack is responsible for deploying out Elastic Filesystems so that the `Fargate` containers can share their file systems and can mount volumes for persistent data. Both `elasticsearch` and `minio` are leveraging this.
- `postgres`
  - This CloudFormation Stack is responsible for deploying out a `RDS` instance running `Postgres`.
- `cluster`
  - This CloudFormation Stack is responsible for deploying out an `ECS` Cluster that will hold and group all `Fargate` services and tasks that make up ReportPortal.
- `migrations`
  - This CloudFormation Stack is responsible for deploying out a `Fargate` task that pulls the `db-scripts` image. Also, this Stack has a `Lambda` function that will invoke/run the `db-scripts` task only once to populate the `Postgres` DB instance.
- `alb`
  - This CloudFormation Stack is responsible for deploying out an `ALB` that has multiple target groups for the UI and API services (`api`, `jobs`, and `uat`) with the appropriate path-pattern rules and health-checks. For example, the `api` listener rule is looking for the path-pattern of `/api` and will forward traffic to the `api` target group which is on port 8585 and runs a health-check on the`/health` endpoint (all the other services are very similar with the exception of the UI as this is the default forwarding action). The `ALB` allows us to communicate with ReportPortal internally through the browser for the UI and through integration (i.e. Playwright) for API calls.
- `app`
  - This CloudFormation Stack is responsible for deploying out all the `Fargate` services and tasks that all make up ReportPortal. Most of the `Fargate` services and tasks are configured very similarly to the `docker-compose`.
 
Currently, all the services and tasks are up and running and healthy; however, the UI is stuck at the `Loading...` screen, which tells me there is something that is either incorrect with the application and/or the infrastructure setup.

⚠️ I decided to not use `traefik` and `index` services as I am using an `ALB` and `Namespace` for service discovery and container communication.

**Screenshots**
- `Loading...` screen issue:
  ![reportportal-ui-loading-screen-stuck](https://github.com/reportportal/reportportal/assets/108296923/a4c8724f-3f44-4285-bf60-de42e4d8d862)
- `Fargate` services all healthy:
![reportportal-service-tasks-fargate](https://github.com/reportportal/reportportal/assets/108296923/fa1ce8a8-654d-4c01-8b1e-d403eed42f6e)

**Logs**
I removed some of the noise surrounding the warnings/errors found in the logs. Here is a list of warnings/errors. I know that some of these logs are not related; however, it might be better to put as much information here as possible:
- `Analyzer`
    - WARNING: you are running uWSGI as root !!! (use the --uid flag)
    - /venv/lib/python3.7/site-packages/sklearn/linear_model/least_angle.py:1442: DeprecationWarning: `np.float` is a deprecated alias for the builtin `float`. To silence this warning, use `float` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.float64` here.
- `API`
  - Unable to create an RSA verifier from verifierKey (ignoreable if using MAC)
- `Elasticsearch`
  - JVM arguments [-Xshare:auto, -Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -XX:+ShowCodeDetailsInExceptionMessages, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Dlog4j2.formatMsgNoLookups=true, -Djava.locale.providers=SPI,COMPAT, --add-opens=java.base/java.io=ALL-UNNAMED, -XX:+UseG1GC, -Djava.io.tmpdir=/tmp/elasticsearch-12671927430012896513, -XX:+HeapDumpOnOutOfMemoryError, -XX:+ExitOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Des.cgroups.hierarchy.override=/, -Xms512m, -Xmx512m, -XX:MaxDirectMemorySize=268435456, -XX:G1HeapRegionSize=4m, -XX:InitiatingHeapOccupancyPercent=30, -XX:G1ReservePercent=15, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/usr/share/elasticsearch/config, -Des.distribution.flavor=default, -Des.distribution.type=docker, -Des.bundled_jdk=true]
  - max virtual memory areas vm.max_map_count [65530] is too low, increase to at least [262144]
- `Metrics`
   - WARNING: you are running uWSGI as root !!! (use the --uid flag)
- `RabbitMQ`
  - [error] <0.1900.0> closing AMQP connection <0.1900.0>
  - [warning] <0.709.0> Message store "CCCKRGUCI5BC4D7Q237IU7AY5/msg_store_persistent": rebuilding indices from scratch
- `UAT`
  - 2023-07-12 20:56:52.370 ERROR 1 --- [ main] c.e.t.r.f.distributed.s3.S3DataStore : Unable to find file 'keystore/secret-integration-salt'
  - Unable to create an RSA verifier from verifierKey (ignoreable if using MAC)

**Versions:**
 - Linux/X86_64, Chrome and Firefox (Loading screen issue shows for both browsers)
 - Not able to get to the login screen. However, here are all the image versions I am using (all `5.7.4`, except metrics which is `5.7.3`):
   - reportportal/service-auto-analyzer:5.7.4
   - reportportal/service-auto-analyzer:5.7.4
   - reportportal/service-api:5.7.4
   - reportportal/service-jobs:5.7.4
   - reportportal/service-metrics-gatherer:5.7.3
   - minio/minio:RELEASE.2020-10-27T04-03-55Z
   - rabbitmq:3.7.16-management
   - reportportal/service-authorization:5.7.4
   - reportportal/service-ui:5.7.4

Please, let me know if I need to provide anything else.
Any feedback would be greatly appreciated.

Thanks!

**Comment by jpaxtonrs** [comment]
Following up on this issue. Thanks!

**Comment by DzmitryHumianiuk** [comment]
@jpaxtonrs looks like web app (JS bundle in UI service) can't see the API service for initial calls to represent the login screen.

**Comment by KenKingson** [comment]
Hey, 
We are facing the same issue, usually this state comes after the night, when the developer tries to open the report portal in the morning, he gets this endless loading. 
The only workaround we found was to restart the containers.

**Comment by DzmitryHumianiuk** [comment]
@KenKingson could you please examine the logs in the `service-api` container prior to restart? 
This should shed some light on the issue leading up to the service being unavailable.

**Comment by MikitaKarbanovich** [comment]
Hi everyone. @DzmitryHumianiuk I was faced with the same issue. Restarts help to fix it, but only for a few hours. I use the latest RP version. Before updating to 23.1 everything was good, but after update we have such behavior.  

Endless logs in the `service-api` :

`2023-08-05 04:08:07   ... 31 common frames omitted
2023-08-05 04:08:07 Caused by: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30001ms.
2023-08-05 04:08:07   at com.zaxxer.hikari.pool.HikariPool.createTimeoutException(HikariPool.java:696)
2023-08-05 04:08:07   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:197)
2023-08-05 04:08:07   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:162)
2023-08-05 04:08:07   at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:100)
2023-08-05 04:08:07   at org.hibernate.engine.jdbc.connections.internal.DatasourceConnectionProviderImpl.getConnection(DatasourceConnectionProviderImpl.java:122)
2023-08-05 04:08:07   at org.hibernate.internal.NonContextualJdbcConnectionAccess.obtainConnection(NonContextualJdbcConnectionAccess.java:38)
2023-08-05 04:08:07   at org.hibernate.resource.jdbc.internal.LogicalConnectionManagedImpl.acquireConnectionIfNeeded(LogicalConnectionManagedImpl.java:108)
2023-08-05 04:08:07   ... 35 common frames omitted
2023-08-05 04:08:07
2023-08-05 04:09:00 2023-08-05 01:09:00.684 ERROR 1 --- [_MisfireHandler] o.s.s.quartz.LocalDataSourceJobStore   : MisfireHandler: Error handling misfires: Failed to obtain DB connection from data source ‘springNonTxDataSource.reportportal’: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30000ms.
2023-08-05 04:09:00
2023-08-05 04:09:00 org.quartz.JobPersistenceException: Failed to obtain DB connection from data source ‘springNonTxDataSource.reportportal’: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30000ms.
2023-08-05 04:09:00   at org.quartz.impl.jdbcjobstore.JobStoreCMT.getNonManagedTXConnection(JobStoreCMT.java:168)
2023-08-05 04:09:00   at org.quartz.impl.jdbcjobstore.JobStoreSupport.doRecoverMisfires(JobStoreSupport.java:3246)
2023-08-05 04:09:00   at org.quartz.impl.jdbcjobstore.JobStoreSupport$MisfireHandler.manage(JobStoreSupport.java:4012)
2023-08-05 04:09:00   at org.quartz.impl.jdbcjobstore.JobStoreSupport$MisfireHandler.run(JobStoreSupport.java:4033)
2023-08-05 04:09:00 Caused by: java.sql.SQLTransientConnectionException: HikariPool-1 - Connection is not available, request timed out after 30000ms.
2023-08-05 04:09:00   at com.zaxxer.hikari.pool.HikariPool.createTimeoutException(HikariPool.java:696)
2023-08-05 04:09:00   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:197)
2023-08-05 04:09:00   at com.zaxxer.hikari.pool.HikariPool.getConnection(HikariPool.java:162)
2023-08-05 04:09:00   at com.zaxxer.hikari.HikariDataSource.getConnection(HikariDataSource.java:100)
2023-08-05 04:09:00   at org.springframework.scheduling.quartz.LocalDataSourceJobStore$2.getConnection(LocalDataSourceJobStore.java:140)
2023-08-05 04:09:00   at org.quartz.utils.DBConnectionManager.getConnection(DBConnectionManager.java:108)
2023-08-05 04:09:00   at org.quartz.impl.jdbcjobstore.JobStoreCMT.getNonManagedTXConnection(JobStoreCMT.java:165)
2023-08-05 04:09:00   ... 3 common frames omitted
2023-08-05 04:09:00
2023-08-05 04:09:35 2023-08-05 01:09:35.305 WARN 1 --- [ult-task-sched3] o.h.engine.jdbc.spi.SqlExceptionHelper  : SQL Error: 0, SQLState: null
2023-08-05 04:09:35 2023-08-05 01:09:35.305 ERROR 1 --- [ult-task-sched3] o.h.engine.jdbc.spi.SqlExceptionHelper  : HikariPool-1 - Connection is not available, request timed out after 30001ms.
2023-08-05 04:09:35 2023-08-05 01:09:35.305 ERROR 1 --- [ult-task-sched3] o.s.s.s.TaskUtils$LoggingErrorHandler  : Unexpected error occurred in scheduled task
2023-08-05 04:09:35
2023-08-05 04:09:35 org.springframework.transaction.CannotCreateTransactionException: Could not open JPA EntityManager for transaction; nested exception is org.hibernate.exception.JDBCConnectionException: Unable to acquire JDBC Connection
2023-08-05 04:09:35   at org.springframework.orm.jpa.JpaTransactionManager.doBegin(JpaTransactionManager.java:467)
2023-08-05 04:09:35   at org.springframework.transaction.support.AbstractPlatformTransactionManager.startTransaction(AbstractPlatformTransactionManag`

**Comment by irynakozak2** [comment]
@jpaxtonrs @MikitaKarbanovich Hello everyone, could you please share the total amount of PostgreSQL connections by executing the following query while connecting to the DB:

`show max_connections;`

Please also take a look at this documentation regarding [API connection pool tuning](https://reportportal.io/docs/installation-steps/OptimalPerformanceHardwareSetup#6-application-connections-pool-tuning) and try to follow the instructions and increase the connection pool for the API service, it could be a quick solution for that issue.

Waiting for your reply.

**Comment by DzmitryHumianiuk** [comment]
@jpaxtonrs and `SELECT * FROM pg_stat_activity;` ??

**Comment by yevheniilavrenchuk** [comment]
For some reason service-api stops communication with rebitmq as well:
```
2023-08-09 11:44:17.385 ERROR 1 --- [-listener-exec4] .a.i.SimpleAsyncUncaughtExceptionHandler : Unexpected exception occurred invoking async method: public void com.epam.ta.reportportal.core.events.listener.LaunchFinishedEventListener.onApplicationEvent(com.epam.ta.reportportal.core.events.activity.LaunchFinishedEvent)

com.epam.ta.reportportal.exception.ReportPortalException: com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<channel.open>(out-of-band=) on channel AMQChannel(amqp://rabbitmq@172.18.0.8:5672/analyzer,5) (#5) timed out
        at com.epam.ta.reportportal.core.analyzer.auto.impl.LogIndexerService.indexLaunchLogs(LogIndexerService.java:109)
        at com.epam.ta.reportportal.core.analyzer.auto.impl.LogIndexerService$$FastClassBySpringCGLIB$$f53c9b3b.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698)
        at com.epam.ta.reportportal.core.analyzer.auto.impl.LogIndexerService$$EnhancerBySpringCGLIB$$c843e7cf.indexLaunchLogs(<generated>)
        at com.epam.ta.reportportal.core.analyzer.auto.starter.decorator.IndexingAutoAnalysisStarter.start(IndexingAutoAnalysisStarter.java:46)
        at com.epam.ta.reportportal.core.analyzer.auto.starter.decorator.IndexingAutoAnalysisStarter$$FastClassBySpringCGLIB$$6c5f0b93.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
        at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
        at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
        at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
        at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:698)
        at com.epam.ta.reportportal.core.analyzer.auto.starter.decorator.IndexingAutoAnalysisStarter$$EnhancerBySpringCGLIB$$7c282fd7.start(<generated>)
        at com.epam.ta.reportportal.core.analyzer.auto.starter.decorator.ExistingAnalyzerStarter.start(ExistingAnalyzerStarter.java:46)
        at com.epam.ta.reportportal.core.events.handler.launch.LaunchAutoAnalysisRunner.handle(LaunchAutoAnalysisRunner.java:51)
        at com.epam.ta.reportportal.core.events.handler.launch.LaunchAutoAnalysisRunner.handle(LaunchAutoAnalysisRunner.java:34)
        at com.epam.ta.reportportal.core.events.subscriber.impl.delegate.ProjectConfigDelegatingSubscriber.lambda$handleEvent$0(ProjectConfigDelegatingSubscriber.java:44)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at com.epam.ta.reportportal.core.events.subscriber.impl.delegate.ProjectConfigDelegatingSubscriber.handleEvent(ProjectConfigDelegatingSubscriber.java:44)
        at com.epam.ta.reportportal.core.events.subscriber.impl.delegate.ProjectConfigDelegatingSubscriber.handleEvent(ProjectConfigDelegatingSubscriber.java:30)
        at com.epam.ta.reportportal.core.events.listener.LaunchFinishedEventListener.lambda$onApplicationEvent$0(LaunchFinishedEventListener.java:45)
        at java.base/java.lang.Iterable.forEach(Iterable.java:75)
        at com.epam.ta.reportportal.core.events.listener.LaunchFinishedEventListener.onApplicationEvent(LaunchFinishedEventListener.java:45)
        at com.epam.ta.reportportal.core.events.listener.LaunchFinishedEventListener$$FastClassBySpringCGLIB$$81f8adcc.invoke(<generated>)
        at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:783)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:753)
        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:115)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by MikitaKarbanovich** [comment]
Hello @irynakozak2. Unfortunately cannot provide this information anymore as I fully reinstalled RP as such behaviour was a blocker for me and I didn't have time to investigate it fully. But the good news - after full reinstallation RP works well.

**Comment by irynakozak2** [comment]
@MikitaKarbanovich Thank you for the information and updates. We will check with the Back End team as we discovered this behavior was reproducible occasionally in the local environment (Docker installation). In our Kubernetes environments, we haven't seen that anywhere.

**Comment by irynakozak2** [comment]
@jpaxtonrs Thank you. I assume the issue can be with the default connection pool initialization or the lack of the default connections count (27). However, it's difficult to prove without understanding what sessions/queries in the database are running and acquiring the connections. Please follow the instructions in the documentation mentioned in the comment above. I assume the issue will be resolved after increasing the default connection pool for the API.

**Comment by DzmitryHumianiuk** [comment]
@irynakozak2 it seems like raising the default values in the default setup we distribute could be a sensible decision. 
Are there any potential downsides or drawbacks to this change that we should consider?

**Comment by luckhiem** [comment]
I still facing the same issue "UI still loading" and Couldn't start RabbitMQ in background. 
```2023-08-10 10:20:56 rabbitmq 03:20:56.61 
2023-08-10 10:20:56 rabbitmq 03:20:56.68 Welcome to the Bitnami rabbitmq container
2023-08-10 10:20:56 rabbitmq 03:20:56.71 Subscribe to project updates by watching https://github.com/bitnami/containers
2023-08-10 10:20:56 rabbitmq 03:20:56.76 Submit issues and feature requests at https://github.com/bitnami/containers/issues
2023-08-10 10:20:56 rabbitmq 03:20:56.80 
2023-08-10 10:20:56 rabbitmq 03:20:56.83 INFO  ==> ** Starting RabbitMQ setup **
2023-08-10 10:20:57 rabbitmq 03:20:57.10 INFO  ==> Validating settings in RABBITMQ_* env vars..
2023-08-10 10:20:57 rabbitmq 03:20:57.60 INFO  ==> Initializing RabbitMQ...
2023-08-10 10:20:57 rabbitmq 03:20:57.94 INFO  ==> Generating random cookie
2023-08-10 10:20:58 rabbitmq 03:20:58.52 INFO  ==> Starting RabbitMQ in background...
2023-08-10 10:20:59 /opt/bitnami/scripts/libos.sh: line 344:   148 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:21:10 /opt/bitnami/scripts/libos.sh: line 344:   246 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:21:21 /opt/bitnami/scripts/libos.sh: line 344:   294 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:21:31 /opt/bitnami/scripts/libos.sh: line 344:   391 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:21:42 /opt/bitnami/scripts/libos.sh: line 344:   439 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:21:53 /opt/bitnami/scripts/libos.sh: line 344:   487 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:22:03 /opt/bitnami/scripts/libos.sh: line 344:   584 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:22:14 /opt/bitnami/scripts/libos.sh: line 344:   632 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:22:25 /opt/bitnami/scripts/libos.sh: line 344:   680 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:22:35 /opt/bitnami/scripts/libos.sh: line 344:   777 Segmentation fault      "$@" > /dev/null 2>&1
2023-08-10 10:22:45 rabbitmq 03:22:45.71 ERROR ==> Couldn't start RabbitMQ in background.

**Comment by DzmitryHumianiuk** [comment]
@luckhiem did you raise the default connections parameter?

**Comment by luckhiem** [comment]
@DzmitryHumianiuk How can we raise the default connections parameter? Please help

**Comment by yevheniilavrenchuk** [comment]
Guys, faced the same thing today. 

Suddenly service-api starting to write these logs and never restarts itself, but marked as unhealthy. Rabbitmq on the other hand is healthy.
```
2023-08-11 06:29:35.183  WARN 1 --- [io-8585-exec-35] o.s.b.a.amqp.RabbitHealthIndicator       : Rabbit health check failed

org.springframework.amqp.AmqpIOException: com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<channel.open>(out-of-band=) on channel AMQChannel(amqp://rabbitmq@172.18.0.5:5672/analyzer,192) (#192) timed out

```

After service-api restart, it successfully connects to rebbitmq:

```
2023-08-11 06:31:03.571  INFO 1 --- [           main] o.s.a.r.c.CachingConnectionFactory       : Attempting to connect to: rabbitmq:5672
2023-08-11 06:31:03.699  INFO 1 --- [           main] o.s.a.r.c.CachingConnectionFactory       : Created new connection: connectionFactory#6b35b0f4:0/SimpleConnection@5292d417 [delegate=amqp://rabbitmq@172.18.0.5:5672/, localPort= 53186]
2023-08-11 06:31:03.966  INFO 1 --- [           main] o.s.amqp.rabbit.core.RabbitAdmin         : Auto-declaring a non-durable or auto-delete Exchange (broadcast.events) durable:false, auto-delete:false. It will be deleted by the broker if it shuts down, and can be redeclared by closing and reopening the connection.
2023-08-11 06:31:03.967  INFO 1 --- [           main] o.s.amqp.rabbit.core.RabbitAdmin         : Auto-declaring a non-durable, auto-delete, or exclusive Queue (broadcast.events.xcGrhtJWRjSJR8wUJllSKg) durable:false, auto-delete:true, exclusive:true. It will be redeclared if the broker stops and is restarted while the connection factory is alive, but all messages will be lost.
2023-08-11 06:31:04.049  INFO 1 --- [           main] c.e.t.r.c.c.r.ReportingConfiguration     : Consumer is created, current consumers count is 1
2023-08-11 06:31:04.050  INFO 1 --- [           main] c.e.t.r.c.c.r.ReportingConfiguration     : Consumer is created, current consumers count is 2
```

**Comment by irynakozak2** [comment]
@luckhiem Hi, could you please specify the CPU and RAM resources for the API service?
And regarding raising the default connection pool, please check [this link](https://reportportal.io/docs/installation-steps/OptimalPerformanceHardwareSetup/#6-application-connections-pool-tuning).
Basically, you should add the environment variable RP_DATASOURCE_MAXIMUMPOOLSIZE=100 for the service-API and check that the PostgreSQL database has more max_connections.

> I still facing the same issue "UI still loading" and Couldn't start RabbitMQ in background.
> 
> ```
> 2023-08-10 10:20:56 rabbitmq 03:20:56.68 Welcome to the Bitnami rabbitmq container
> 2023-08-10 10:20:56 rabbitmq 03:20:56.71 Subscribe to project updates by watching https://github.com/bitnami/containers
> 2023-08-10 10:20:56 rabbitmq 03:20:56.76 Submit issues and feature requests at https://github.com/bitnami/containers/issues
> 2023-08-10 10:20:56 rabbitmq 03:20:56.80 
> 2023-08-10 10:20:56 rabbitmq 03:20:56.83 INFO  ==> ** Starting RabbitMQ setup **
> 2023-08-10 10:20:57 rabbitmq 03:20:57.10 INFO  ==> Validating settings in RABBITMQ_* env vars..
> 2023-08-10 10:20:57 rabbitmq 03:20:57.60 INFO  ==> Initializing RabbitMQ...
> 2023-08-10 10:20:57 rabbitmq 03:20:57.94 INFO  ==> Generating random cookie
> 2023-08-10 10:20:58 rabbitmq 03:20:58.52 INFO  ==> Starting RabbitMQ in background...
> 2023-08-10 10:20:59 /opt/bitnami/scripts/libos.sh: line 344:   148 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:21:10 /opt/bitnami/scripts/libos.sh: line 344:   246 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:21:21 /opt/bitnami/scripts/libos.sh: line 344:   294 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:21:31 /opt/bitnami/scripts/libos.sh: line 344:   391 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:21:42 /opt/bitnami/scripts/libos.sh: line 344:   439 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:21:53 /opt/bitnami/scripts/libos.sh: line 344:   487 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:22:03 /opt/bitnami/scripts/libos.sh: line 344:   584 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:22:14 /opt/bitnami/scripts/libos.sh: line 344:   632 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:22:25 /opt/bitnami/scripts/libos.sh: line 344:   680 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:22:35 /opt/bitnami/scripts/libos.sh: line 344:   777 Segmentation fault      "$@" > /dev/null 2>&1
> 2023-08-10 10:22:45 rabbitmq 03:22:45.71 ERROR ==> Couldn't start RabbitMQ in background.
> ```

**Comment by jpaxtonrs** [comment]
> @jpaxtonrs and `SELECT * FROM pg_stat_activity;` ??



datid | datname | pid | usesysid | usename | application_name | client_addr | client_hostname | client_port | backend_start | xact_start | query_start | state_change | wait_event_type | wait_event | state | backend_xid | backend_xmin | query | backend_type
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
  |   | 395 | 10 | rdsadmin |   |   |   |   | 2023-06-30 02:33:41.568806+00 |   |   |   | Activity | LogicalLauncherMain |   |   |   |   | logical replication launcher
  |   | 392 |   |   |   |   |   |   | 2023-06-30 02:33:41.569456+00 |   |   |   | Activity | AutoVacuumMain |   |   |   |   | autovacuum launcher
16384 | rdsadmin | 10801 | 10 | rdsadmin | PostgreSQL JDBC Driver |   |   | -1 | 2023-07-29 05:34:34.956147+00 |   | 2023-08-11 14:55:31.644025+00 | 2023-08-11 14:55:31.644054+00 | Client | ClientRead | idle |   |   | SELECT value FROM rds_heartbeat2 | client backend
16396 | reportportal | 30938 | 16394 | reportportal | TablePlus | XXX.XX.XX.XXX |   | 52937 | 2023-08-11 14:54:43.729941+00 | 2023-08-11 14:55:39.2058+00 | 2023-08-11 14:55:39.2058+00 | 2023-08-11 14:55:39.205804+00 |   |   | active |   | 73252 | SELECT * FROM pg_stat_activity; | client backend
  |   | 390 |   |   |   |   |   |   | 2023-06-30 02:33:41.567964+00 |   |   |   | Activity | BgWriterHibernate |   |   |   |   | background writer
  |   | 389 |   |   |   |   |   |   | 2023-06-30 02:33:41.56744+00 |   |   |   | Activity | CheckpointerMain |   |   |   |   | checkpointer
  |   | 391 |   |   |   |   |   |   | 2023-06-30 02:33:41.570068+00 |   |   |   | Activity | WalWriterMain |   |   |   |   | walwriter

**Comment by irynakozak2** [comment]
Seems like there is a normal state of the PG activities and no issues with lots of idle connections from the ReportPortal. Did you manage to fix the issue after the redeploying CloudFormation stack?

> > @jpaxtonrs and `SELECT * FROM pg_stat_activity;` ??
> 
> datid	datname	pid	usesysid	usename	application_name	client_addr	client_hostname	client_port	backend_start	xact_start	query_start	state_change	wait_event_type	wait_event	state	backend_xid	backend_xmin	query	backend_type
>  	 	395	10	rdsadmin	 	 	 	 	2023-06-30 02:33:41.568806+00	 	 	 	Activity	LogicalLauncherMain	 	 	 	 	logical replication launcher
>  	 	392	 	 	 	 	 	 	2023-06-30 02:33:41.569456+00	 	 	 	Activity	AutoVacuumMain	 	 	 	 	autovacuum launcher
> 16384	rdsadmin	10801	10	rdsadmin	PostgreSQL JDBC Driver	 	 	-1	2023-07-29 05:34:34.956147+00	 	2023-08-11 14:55:31.644025+00	2023-08-11 14:55:31.644054+00	Client	ClientRead	idle	 	 	SELECT value FROM rds_heartbeat2	client backend
> 16396	reportportal	30938	16394	reportportal	TablePlus	XXX.XX.XX.XXX	 	52937	2023-08-11 14:54:43.729941+00	2023-08-11 14:55:39.2058+00	2023-08-11 14:55:39.2058+00	2023-08-11 14:55:39.205804+00	 	 	active	 	73252	SELECT * FROM pg_stat_activity;	client backend
>  	 	390	 	 	 	 	 	 	2023-06-30 02:33:41.567964+00	 	 	 	Activity	BgWriterHibernate	 	 	 	 	background writer
>  	 	389	 	 	 	 	 	 	2023-06-30 02:33:41.56744+00	 	 	 	Activity	CheckpointerMain	 	 	 	 	checkpointer
>  	 	391	 	 	 	 	 	 	2023-06-30 02:33:41.570068+00	 	 	 	Activity	WalWriterMain	 	 	 	 	walwriter

**Comment by irynakozak2** [comment]
@KenKingson @MikitaKarbanovich @luckhiem @jpaxtonrs 
Please check the Network tab in your browser on the requests while opening the login page.
I'm currently debugging the issue and didn't manage to reproduce it locally with docker-compose (with connections), but I faced almost the same behavior on other running instances.

**Comment by KenKingson** [comment]
@irynakozak2 here is the screenshot 
![image](https://github.com/reportportal/reportportal/assets/22968831/c28652af-3d38-40fb-8c26-18502a27043b)

**Comment by DzmitryHumianiuk** [comment]
@traj279 good to know that it works.
but it seems to be a very beefy setup for the tool :/

**Comment by jpaxtonrs** [comment]
> Seems like there is a normal state of the PG activities and no issues or lots of idle connections from the ReportPortal. Did you manage to fix the issue after the redeploying CloudFormation stack?
> 
> > > @jpaxtonrs and `SELECT * FROM pg_stat_activity;` ??
> > 
> > 
> > datid	datname	pid	usesysid	usename	application_name	client_addr	client_hostname	client_port	backend_start	xact_start	query_start	state_change	wait_event_type	wait_event	state	backend_xid	backend_xmin	query	backend_type
> >  	 	395	10	rdsadmin	 	 	 	 	2023-06-30 02:33:41.568806+00	 	 	 	Activity	LogicalLauncherMain	 	 	 	 	logical replication launcher
> >  	 	392	 	 	 	 	 	 	2023-06-30 02:33:41.569456+00	 	 	 	Activity	AutoVacuumMain	 	 	 	 	autovacuum launcher
> > 16384	rdsadmin	10801	10	rdsadmin	PostgreSQL JDBC Driver	 	 	-1	2023-07-29 05:34:34.956147+00	 	2023-08-11 14:55:31.644025+00	2023-08-11 14:55:31.644054+00	Client	ClientRead	idle	 	 	SELECT value FROM rds_heartbeat2	client backend
> > 16396	reportportal	30938	16394	reportportal	TablePlus	XXX.XX.XX.XXX	 	52937	2023-08-11 14:54:43.729941+00	2023-08-11 14:55:39.2058+00	2023-08-11 14:55:39.2058+00	2023-08-11 14:55:39.205804+00	 	 	active	 	73252	SELECT * FROM pg_stat_activity;	client backend
> >  	 	390	 	 	 	 	 	 	2023-06-30 02:33:41.567964+00	 	 	 	Activity	BgWriterHibernate	 	 	 	 	background writer
> >  	 	389	 	 	 	 	 	 	2023-06-30 02:33:41.56744+00	 	 	 	Activity	CheckpointerMain	 	 	 	 	checkpointer
> >  	 	391	 	 	 	 	 	 	2023-06-30 02:33:41.570068+00	 	 	 	Activity	WalWriterMain	 	 	 	 	walwriter

No, I ran into the same issue. I'm still thinking that there is some sort of networking issue. It's just not showing in the logs.
Is there anyway that we can add additional debug log lines whenever this problem occurs?

**Comment by venom0107** [comment]
@jpaxtonrs @irynakozak2 I'm also facing same issue. Have tried restarting service-api multiple times. After restarting the service-api container, everything works well. But as soon as we start sending data from our automation tests, after some 300-400 cases execution, the service-api starts responding with 405's. And then the report portal UI gets stuck on `Loading...` page.

The EC2 instance we are using is t3a.xlarge [ t3a.xlarge | CPU - 4 | RAM - 16.0 ]
Here's my max connection: 
```reportdb=> show max_connections;
 max_connections
-----------------
 821
(1 row)
```

And here's my response for `SELECT * FROM pg_stat_activity;`
```
 datid | datname  |  pid  | leader_pid | usesysid |  usename   |    application_name    | client_addr | client_hostname | client_port |         backend_start   |          xact_start           |          query_start          |         state_change          | wait_event_type |     wait_event      |        state        |backend_xid | backend_xmin |    query                                                                                                                                                    |         backend_type
-------+----------+-------+------------+----------+------------+------------------------+-------------+-----------------+-------------+-------------------------------+-------------------------------+-------------------------------+-------------------------------+-----------------+---------------------+---------------------+-------------+--------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------
       |          |   397 |            |       10 | rdsadmin   |                        |             |                 |             | 2023-10-04 11:57:19.620348+00 |                               |                               |                               | Activity        | LogicalLauncherMain |                     |            |              |                                                                                                                                                    | logical replication launcher
       |          |   394 |            |          |            |                        |             |                 |             | 2023-10-04 11:57:19.624208+00 |                               |                               |                               | Activity        | AutoVacuumMain      |                     |            |              |                                                                                                                                                    | autovacuum launcher
 16398 | reportdb | 17375 |            |    16396 | reportuser | PostgreSQL JDBC Driver | 10.7.6.189  |                 |       51730 | 2023-10-04 16:11:38.532594+00 |                               | 2023-10-04 16:11:38.543139+00 | 2023-10-04 16:11:38.543152+00 | Client          | ClientRead          | idle                |            |              | SET application_name = 'PostgreSQL JDBC Driver'                                                                                                                                                    | client backend
 16398 | reportdb | 17353 |            |    16396 | reportuser | PostgreSQL JDBC Driver | 10.7.6.189  |                 |       54864 | 2023-10-04 16:11:14.934408+00 |                               | 2023-10-04 16:11:14.944382+00 | 2023-10-04 16:11:14.944395+00 | Client          | ClientRead          | idle                |            |              | SET application_name = 'PostgreSQL JDBC Driver'                                                                                                                                                    | client backend
 16398 | reportdb | 17287 |            |    16396 | reportuser | PostgreSQL JDBC Driver | 10.7.6.189  |                 |       54944 | 2023-10-04 16:10:07.68402+00  |                               | 2023-10-04 16:35:10.382199+00 | 2023-10-04 16:35:10.382203+00 | Client          | ClientRead          | idle                |            |              | ```

**Comment by venom0107** [comment]
Also attaching result for `docker ps` command, for reference, just in case:

<img width="1478" alt="Screenshot 2023-10-04 at 10 31 55 PM" src="https://github.com/reportportal/reportportal/assets/49713824/4752f683-11b9-458f-b7f5-6b3df44fc915">

**Comment by venom0107** [comment]
Also attaching result for docker ps command, for reference, just in case:
<img width="1377" alt="Screenshot 2023-10-04 at 10 33 50 PM" src="https://github.com/reportportal/reportportal/assets/49713824/b51de052-8f03-410a-a1a0-6c09fd3a3cdf">

**Comment by DzmitryHumianiuk** [comment]
@venom0107 could you please try latest 23.2 release.

Just delete everything and start from scratch.
Please also make sure that you have at least 6GB allocated for the docker machine.

**Comment by DzmitryHumianiuk** [comment]
If this still remains a concern or if there are new developments, please feel free to reopen or create a new issue. 

Thank you for your understanding and contributions to the project!

**Comment by pankajrawat1996** [comment]
@DzmitryHumianiuk @venom0107 I am facing same issue API service is intermittently going unhealthy then I have to restart the service. below are the logs


2024-09-29 06:36:54.093 ERROR 1 --- [io-8585-exec-27] c.e.r.r.c.e.rest.DefaultErrorResolver    : com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<queue.declare>(ticket=0, queue=amq.rabbitmq.reply-to, passive=true, durable=false, exclusive=true, auto-delete=true, nowait=false, arguments=null) on channel AMQChannel(amqp://rabbitmq@172.23.0.5:5672/analyzer,1) (#1) timed out

org.springframework.amqp.AmqpIOException: com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<queue.declare>(ticket=0, queue=amq.rabbitmq.reply-to, passive=true, durable=false, exclusive=true, auto-delete=true, nowait=false, arguments=null) on channel AMQChannel(amqp://rabbitmq@172.23.0.5:5672/analyzer,1) (#1) timed out
	at org.springframework.amqp.rabbit.support.RabbitExceptionTranslator.convertRabbitAccessException(RabbitExceptionTranslator.java:70)
	at org.springframework.amqp.rabbit.connection.RabbitAccessor.convertRabbitAccessException(RabbitAccessor.java:113)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.doExecute(RabbitTemplate.java:2202)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.execute(RabbitTemplate.java:2148)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.execute(RabbitTemplate.java:2128)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.useDirectReplyTo(RabbitTemplate.java:1018)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.evaluateFastReplyTo(RabbitTemplate.java:990)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.doSendAndReceive(RabbitTemplate.java:1844)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.convertSendAndReceiveRaw(RabbitTemplate.java:1818)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.convertSendAndReceiveAsType(RabbitTemplate.java:1788)
	at org.springframework.amqp.rabbit.core.RabbitOperations.convertSendAndReceiveAsType(RabbitOperations.java:371)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.convertSendAndReceiveAsType(RabbitTemplate.java:1730)
	at com.epam.ta.reportportal.core.analyzer.auto.client.impl.AnalyzerServiceClientImpl.searchLogs(AnalyzerServiceClientImpl.java:98)
	at com.epam.ta.reportportal.core.analyzer.auto.impl.SearchLogServiceImpl.processRequest(SearchLogServiceImpl.java:156)
	at com.epam.ta.reportportal.core.analyzer.auto.impl.SearchLogServiceImpl.lambda$search$3(SearchLogServiceImpl.java:118)
	at java.base/java.util.Optional.map(Optional.java:260)
	at com.epam.ta.reportportal.core.analyzer.auto.impl.SearchLogServiceImpl.search(SearchLogServiceImpl.java:117)
	at com.epam.ta.reportportal.core.analyzer.auto.impl.SearchLogServiceImpl$$FastClassBySpringCGLIB$$fb44ce98.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.transaction.interceptor.TransactionInterceptor$1.proceedWithInvocation(TransactionInterceptor.java:123)
	at org.springframework.transaction.interceptor.TransactionAspectSupport.invokeWithinTransaction(TransactionAspectSupport.java:388)
	at org.springframework.transaction.interceptor.TransactionInterceptor.invoke(TransactionInterceptor.java:119)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.epam.ta.reportportal.core.analyzer.auto.impl.SearchLogServiceImpl$$EnhancerBySpringCGLIB$$6e2b3366.search(<generated>)
	at com.epam.ta.reportportal.ws.controller.LogController.searchLogs(LogController.java:309)
	at com.epam.ta.reportportal.ws.controller.LogController$$FastClassBySpringCGLIB$$f7673969.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.security.access.intercept.aopalliance.MethodSecurityInterceptor.invoke(MethodSecurityInterceptor.java:67)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at com.epam.ta.reportportal.ws.controller.LogController$$EnhancerBySpringCGLIB$$b6c7edfa.searchLogs(<generated>)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.web.method.support.InvocableHandlerMethod.doInvoke(InvocableHandlerMethod.java:205)
	at org.springframework.web.method.support.InvocableHandlerMethod.invokeForRequest(InvocableHandlerMethod.java:150)
	at org.springframework.web.servlet.mvc.method.annotation.ServletInvocableHandlerMethod.invokeAndHandle(ServletInvocableHandlerMethod.java:117)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.invokeHandlerMethod(RequestMappingHandlerAdapter.java:903)
	at org.springframework.web.servlet.mvc.method.annotation.RequestMappingHandlerAdapter.handleInternal(RequestMappingHandlerAdapter.java:809)
	at org.springframework.web.servlet.mvc.method.AbstractHandlerMethodAdapter.handle(AbstractHandlerMethodAdapter.java:87)
	at org.springframework.web.servlet.DispatcherServlet.doDispatch(DispatcherServlet.java:1072)
	at org.springframework.web.servlet.DispatcherServlet.doService(DispatcherServlet.java:965)
	at org.springframework.web.servlet.FrameworkServlet.processRequest(FrameworkServlet.java:1006)
	at org.springframework.web.servlet.FrameworkServlet.doPost(FrameworkServlet.java:909)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:555)
	at org.springframework.web.servlet.FrameworkServlet.service(FrameworkServlet.java:883)
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:623)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:209)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:51)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.multipart.support.MultipartFilter.doFilterInternal(MultipartFilter.java:125)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:352)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.invoke(FilterSecurityInterceptor.java:117)
	at org.springframework.security.web.access.intercept.FilterSecurityInterceptor.doFilter(FilterSecurityInterceptor.java:83)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:126)
	at org.springframework.security.web.access.ExceptionTranslationFilter.doFilter(ExceptionTranslationFilter.java:120)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:131)
	at org.springframework.security.web.session.SessionManagementFilter.doFilter(SessionManagementFilter.java:85)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.authentication.AnonymousAuthenticationFilter.doFilter(AnonymousAuthenticationFilter.java:100)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.servletapi.SecurityContextHolderAwareRequestFilter.doFilter(SecurityContextHolderAwareRequestFilter.java:164)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.savedrequest.RequestCacheAwareFilter.doFilter(RequestCacheAwareFilter.java:63)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.oauth2.provider.authentication.OAuth2AuthenticationProcessingFilter.doFilter(OAuth2AuthenticationProcessingFilter.java:182)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:107)
	at org.springframework.security.web.authentication.logout.LogoutFilter.doFilter(LogoutFilter.java:93)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.header.HeaderWriterFilter.doHeadersAfter(HeaderWriterFilter.java:90)
	at org.springframework.security.web.header.HeaderWriterFilter.doFilterInternal(HeaderWriterFilter.java:75)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:117)
	at org.springframework.security.web.context.SecurityContextPersistenceFilter.doFilter(SecurityContextPersistenceFilter.java:87)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.context.request.async.WebAsyncManagerIntegrationFilter.doFilterInternal(WebAsyncManagerIntegrationFilter.java:62)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.session.DisableEncodeUrlFilter.doFilterInternal(DisableEncodeUrlFilter.java:42)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.springframework.security.web.FilterChainProxy$VirtualFilterChain.doFilter(FilterChainProxy.java:361)
	at org.springframework.security.web.FilterChainProxy.doFilterInternal(FilterChainProxy.java:225)
	at org.springframework.security.web.FilterChainProxy.doFilter(FilterChainProxy.java:190)
	at org.springframework.web.filter.DelegatingFilterProxy.invokeDelegate(DelegatingFilterProxy.java:354)
	at org.springframework.web.filter.DelegatingFilterProxy.doFilter(DelegatingFilterProxy.java:267)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.RequestContextFilter.doFilterInternal(RequestContextFilter.java:100)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.FormContentFilter.doFilterInternal(FormContentFilter.java:93)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.boot.actuate.metrics.web.servlet.WebMvcMetricsFilter.doFilterInternal(WebMvcMetricsFilter.java:96)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.springframework.web.filter.CharacterEncodingFilter.doFilterInternal(CharacterEncodingFilter.java:201)
	at org.springframework.web.filter.OncePerRequestFilter.doFilter(OncePerRequestFilter.java:117)
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:178)
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:153)
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:168)
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:90)
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:481)
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:130)
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:93)
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:74)
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:346)
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:390)
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:63)
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:928)
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1794)
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:52)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1191)
	at org.apache.tomcat.util.threads.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:659)
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:63)
	at java.base/java.lang.Thread.run(Thread.java:1583)
Caused by: com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<queue.declare>(ticket=0, queue=amq.rabbitmq.reply-to, passive=true, durable=false, exclusive=true, auto-delete=true, nowait=false, arguments=null) on channel AMQChannel(amqp://rabbitmq@172.23.0.5:5672/analyzer,1) (#1) timed out
	at com.rabbitmq.client.impl.AMQChannel.wrapTimeoutException(AMQChannel.java:313)
	at com.rabbitmq.client.impl.AMQChannel.privateRpc(AMQChannel.java:295)
	at com.rabbitmq.client.impl.AMQChannel.exnWrappingRpc(AMQChannel.java:141)
	at com.rabbitmq.client.impl.ChannelN.queueDeclarePassive(ChannelN.java:1012)
	at com.rabbitmq.client.impl.ChannelN.queueDeclarePassive(ChannelN.java:46)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at org.springframework.amqp.rabbit.connection.CachingConnectionFactory$CachedChannelInvocationHandler.invoke(CachingConnectionFactory.java:1157)
	at jdk.proxy2/jdk.proxy2.$Proxy315.queueDeclarePassive(Unknown Source)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.lambda$useDirectReplyTo$2(RabbitTemplate.java:1019)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.invokeAction(RabbitTemplate.java:2237)
	at org.springframework.amqp.rabbit.core.RabbitTemplate.doExecute(RabbitTemplate.java:2196)
	... 133 common frames omitted
Caused by: java.util.concurrent.TimeoutException: null
	at com.rabbitmq.utility.BlockingCell.get(BlockingCell.java:77)
	at com.rabbitmq.utility.BlockingCell.uninterruptibleGet(BlockingCell.java:120)
	at com.rabbitmq.utility.BlockingValueOrException.uninterruptibleGetValue(BlockingValueOrException.java:36)
	at com.rabbitmq.client.impl.AMQChannel$BlockingRpcContinuation.getReply(AMQChannel.java:502)
	at com.rabbitmq.client.impl.AMQChannel.privateRpc(AMQChannel.java:293)
	... 143 common frames omitted

2024-09-29 06:36:54.094  WARN 1 --- [io-8585-exec-27] c.e.r.r.c.e.rest.RestExceptionHandler    : Resolved [org.springframework.amqp.AmqpIOException: com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<queue.declare>(ticket=0, queue=amq.rabbitmq.reply-to, passive=true, durable=false, exclusive=true, auto-delete=true, nowait=false, arguments=null) on channel AMQChannel(amqp://rabbitmq@172.23.0.5:5672/analyzer,1) (#1) timed out]

**Comment by DzmitryHumianiuk** [comment]
@pankajrawat1996 your rabbitMQ is unreachable.
here is what GPT is saying on this:
___________________________

```

```

The error logs you've provided indicate a timeout issue when your application is attempting to communicate with the RabbitMQ server. Here's a detailed analysis of the problem and potential solutions:

### **Problem Analysis**

**Error Message:**
```
com.rabbitmq.client.ChannelContinuationTimeoutException: Continuation call for method #method<queue.declare>(ticket=0, queue=amq.rabbitmq.reply-to, passive=true, durable=false, exclusive=true, auto-delete=true, nowait=false, arguments=null) on channel AMQChannel(amqp://rabbitmq@172.23.0.5:5672/analyzer,1) (#1) timed out
```

**Key Points:**
1. **Timeout Exception:** The `ChannelContinuationTimeoutException` suggests that the RabbitMQ client did not receive a response within the expected timeframe when trying to declare the `amq.rabbitmq.reply-to` queue.
2. **Passive Queue Declaration:** The `passive=true` parameter means the client is attempting to check if the queue exists without creating it. If the queue doesn't exist, it would typically result in a `ChannelException`, but here it's a timeout, indicating no response.
3. **Connection Details:** The client is trying to connect to RabbitMQ at `172.23.0.5:5672` using the `analyzer` virtual host.

### **Possible Causes**

1. **RabbitMQ Server Unreachable:**
   - The RabbitMQ server might be down or not reachable at the specified IP and port.
   
2. **Network Issues:**
   - Network partitions, firewall rules, or routing issues could prevent the application from reaching the RabbitMQ server.
   
3. **Resource Constraints on RabbitMQ:**
   - The server might be overloaded, lacking sufficient resources (CPU, memory), or hitting limits on channels or connections.
   
4. **Misconfiguration:**
   - Incorrect RabbitMQ connection settings in the application (e.g., wrong hostname, port, virtual host, or credentials).
   
5. **Firewall or Security Groups:**
   - Firewalls or security groups might be blocking traffic to RabbitMQ's port (`5672`).
   
6. **DNS Issues:**
   - If using a hostname instead of an IP, DNS resolution problems could prevent successful connections.
   
7. **Version Incompatibility:**
   - Incompatibility between the RabbitMQ client library version and the RabbitMQ server version.

### **Recommended Solutions**

1. **Verify RabbitMQ Server Status:**
   - **Check if RabbitMQ is Running:**
     - Ensure that the RabbitMQ service is active on `172.23.0.5`.
     - Use commands like `systemctl status rabbitmq-server` or `rabbitmqctl status`.
     
   - **Restart RabbitMQ:**
     - If the server is unresponsive, consider restarting the RabbitMQ service.
     
2. **Test Network Connectivity:**
   - **Ping the Server:**
     - `ping 172.23.0.5` to check basic connectivity.
     
   - **Check Port Accessibility:**
     - Use `telnet 172.23.0.5 5672` or `nc -zv 172.23.0.5 5672` to verify if the port is open.
     
3. **Review Application Configuration:**
   - **Connection Parameters:**
     - Ensure that the hostname/IP, port, virtual host (`analyzer`), username, and password are correctly configured.
     
   - **Timeout Settings:**
     - Consider increasing the timeout settings in your RabbitMQ client configuration to accommodate potential delays.
     
4. **Inspect RabbitMQ Logs:**
   - **Server Logs:**
     - Check RabbitMQ server logs for any errors or warnings that might indicate underlying issues.
     - Logs are typically located at `/var/log/rabbitmq/` or accessible via the RabbitMQ Management UI.
     
5. **Monitor Server Resources:**
   - **Resource Utilization:**
     - Ensure the server has adequate CPU, memory, and disk space.
     
   - **Connection Limits:**
     - RabbitMQ has limits on the number of connections and channels. Verify if these limits are being approached or exceeded.
     
6. **Check Firewall and Security Settings:**
   - **Firewall Rules:**
     - Ensure that firewalls on both the application and RabbitMQ server sides allow traffic on port `5672`.
     
   - **Security Groups:**
     - If hosted in cloud environments (e.g., AWS, Azure), verify that security groups or network ACLs permit the necessary traffic.
     
7. **DNS Verification (if applicable):**
   - **Hostname Resolution:**
     - If using a hostname instead of an IP, ensure that DNS resolves correctly using `nslookup` or `dig`.
     
8. **Update Client Libraries:**
   - **Version Compatibility:**
     - Ensure that the RabbitMQ client library used by your application is compatible with the RabbitMQ server version.
     
   - **Upgrade if Necessary:**
     - Consider updating to the latest stable versions to benefit from fixes and improvements.
     
9. **Handle the `amq.rabbitmq.reply-to` Queue Appropriately:**
   - **Ensure Proper Usage:**
     - The `amq.rabbitmq.reply-to` queue is used for RPC-style communication. Ensure that your application correctly manages this queue.
     
   - **Check for Pre-declaration:**
     - Sometimes, pre-declaring certain queues or ensuring idempotent declarations can help prevent such issues.
     
10. **Enable RabbitMQ Management Plugin (if not already enabled):**
    - **Monitor via UI:**
      - The RabbitMQ Management Plugin provides a web-based UI to monitor queues, connections, channels, and more.
      - Enable it using `rabbitmq-plugins enable rabbitmq_management` and access it typically at `http://172.23.0.5:15672/`.
      
### **Additional Recommendations**

- **Implement Retry Logic:**
  - Incorporate retry mechanisms in your application to handle transient connectivity issues gracefully.
  
- **Use Heartbeats:**
  - Configure heartbeats between the client and RabbitMQ to detect and handle broken connections promptly.
  
- **Load Balancing and High Availability:**
  - If applicable, set up RabbitMQ in a clustered or high-availability configuration to mitigate single points of failure.
  
- **Consult RabbitMQ Documentation:**
  - Refer to the [official RabbitMQ documentation](https://www.rabbitmq.com/documentation.html) for best practices and advanced configurations.

### **Conclusion**

The `ChannelContinuationTimeoutException` typically points to connectivity or responsiveness issues between your application and the RabbitMQ server. By systematically verifying the server's status, network connectivity, application configurations, and resource utilization, you can identify and resolve the root cause of the timeout. Implementing monitoring and resilience strategies will also help prevent similar issues in the future.

**Comment by pankajrawat1996** [comment]
@DzmitryHumianiuk Others are also facing the same issue ..check below link
https://github.com/reportportal/reportportal/issues/2273#issuecomment-2384736877
@akshayamaldhure
